resources/processor-types/add_field.yaml

```
@@ -0,0 +1,204 @@
+apiVersion: bindplane.observiq.com/v1
+kind: ProcessorType
+metadata:
+  name: add_fields
+  displayName: Add Fields
+  description: Add telemetry fields.
+  labels:
+    category: Transform
+spec:
+  version: 0.0.1
+  parameters:
+    - name: telemetry_types
+      label: Enable Telemetry
+      type: enums
+      description: "Select which types of telemetry you'd like to enable this processor for."
+      validValues: ["Metrics", "Logs", "Traces"]
+      default: ["Metrics", "Logs", "Traces"]
+      options:
+        gridColumns: 12
+
+    - name: resource_action
+      label: Resource Action
+      type: enum
+      description: |
+        insert: Add resource attribute if it does not exist.
+        update: Update existing value.
+        upsert: Insert or update.
+      default: upsert
+      validValues:
+        - insert
+        - update
+        - upsert
+
+    - name: resource_attributes
+      label: Resource Attributes
+      description: "List of resource attribute fields to add from telemetry data."
+      type: map
+      options:
+        gridColumns: 12
+
+    - name: attributes_action
+      label: Attribute Action
+      type: enum
+      description: |
+        insert: Add resource attribute if it does not exist.
+        update: Update existing value.
+        upsert: Insert or update.
+      default: upsert
+      validValues:
+        - insert
+        - update
+        - upsert
+
+    - name: attributes
+      label: Attributes
+      type: map
+      options:
+        gridColumns: 12
+
+    - name: body_action
+      label: Body Action
+      type: enum
+      description: |
+        insert: Add resource attribute if it does not exist.
+        update: Update existing value.
+        upsert: Insert or update.
+      default: upsert
+      validValues:
+        - insert
+        - update
+        - upsert
+      relevantIf:
+        - name: telemetry_types
+          operator: containsAny
+          value: ["Logs"]
+
+    - name: body
+      label: Body Fields
+      description: "List of body fields to add to log data."
+      type: map
+      relevantIf:
+        - name: telemetry_types
+          operator: containsAny
+          value: ["Logs"]
+      options:
+        gridColumns: 12
+
+  logs:
+    processors: |
+      {{ if has "Logs" .telemetry_types }}
+        - transform/logs:
+            log_statements:
+            - context: log
```

One last thing here, I think we need to add `error_mode: ignore` to all three of the transform processors
```suggestion
        - transform/logs:
            error_mode: ignore
            log_statements:
            - context: log
```

---

I was observing log records getting dropped if the body wasn't index-able, and I think this is why.